lib/src/main/java/com/nextcloud/android/sso/api/NextcloudAPI.java

```diff
@@ -141,7 +141,10 @@ public <T> T convertStreamToTargetEntity(InputStream inputStream, Type targetEnt
             } else {
                 result = gson.fromJson(targetReader, targetEntity);
                 if (result == null) {
-                    if (targetEntity == Object.class) {
+                    if(targetEntity == Void.class) {
+                        throw new IllegalStateException("Detected Void return type - please " +
+                                "switch to the new return type \"EmptyResponse\" instead.");
+                    } else if (targetEntity == Object.class) {
```

Comment:
Can't we use 

```java
ensureTypeNotVoid(targetEntity);
```

to ensure the same error message?

Reply:
Yes, good point! 👍

Reply:
Actually, if we call `ensureTypeNotVoid` in `performNetworkRequestV2` instead of all the current places, we should easily tackle all cases, no?

Reply:
In theory - yes. But `performNetworkRequestV2` has no response type information thus we can't add the check there.